### PR TITLE
Fix issue #699

### PR DESCRIPTION
**Pull request by Sag0Sag0**

### CODE REVIEW

**Comment by zen3ger** [comment]
The changes caused `cd ../` to fail.
```sh
zen3ger:~/Gitstuff/ion# cd src/
zen3ger:~/Gitstuff/ion/src# dirs -v
 0  src
zen3ger:~/Gitstuff/ion/src# cd ../
ion: failed to set current dir to : No such file or directory (os error 2)
```

Also `cd -` stopped working. It's expected to jump back to the previous directory in history.
```sh
zen3ger:~/Gitstuff/ion# cd src/lib
zen3ger:~/Gitstuff/ion/src/lib# cd -
zen3ger:~/Gitstuff/ion/src/lib# pwd
/home/zen3ger/Gitstuff/ion/src/lib
```
Without these changes `cd -` seems to swap the 0th and 1st entry in the dir stack. (build of master):
```sh
zen3ger:~/Gitstuff/ion# cd src/lib/
zen3ger:~/Gitstuff/ion/src/lib# dirs -v
 0  /home/zen3ger/Gitstuff/ion/src/lib
 1  /home/zen3ger/Gitstuff/ion
zen3ger:~/Gitstuff/ion/src/lib# cd -
/home/zen3ger/Gitstuff/ion
zen3ger:~/Gitstuff/ion# dirs -v
 0  /home/zen3ger/Gitstuff/ion
 1  /home/zen3ger/Gitstuff/ion/src/lib
```

**Comment by Sag0Sag0** [comment]
Thanks, for catching that! I'll fix up the PR in a bit.

**Comment by zen3ger** [comment]
Seems perfect.

File: src/lib/builtins/mod.rs [hunk 1]
```diff
@@ -166,26 +166,27 @@ pub fn builtin_cd(args: &[&str], shell: &mut Shell) -> i32 {
...
-            );
+                Err(_) => env::set_var("PWD", "?"),
+            };
             SUCCESS
         }
         Err(why) => {
             let stderr = io::stderr();
             let mut stderr = stderr.lock();
             let _ = stderr.write_all(why.as_bytes());
+            let _ = stderr.write_all("Problem".as_bytes());
```
Comment (mmstick) [inline src/lib/builtins/mod.rs]: May want to remove this, and altogether we should use `eprintln` instead.

File: src/lib/shell/directory_stack.rs [hunk 2]
```diff
@@ -187,13 +185,33 @@ impl DirectoryStack {
...
                 }
             }
             None => self.switch_to_home_directory(variables),
         }
     }
 
+    fn update_env_variables(&mut self, variables: &mut Variables) {
+        // Update $OLDPWD
+        let old_pwd = variables.get_var_or_empty("PWD");
+        if &old_pwd == "" {
```
Comment (mmstick) [inline src/lib/shell/directory_stack.rs]: Could write `if old_pwd.is_empty()`

File: src/lib/shell/directory_stack.rs [hunk 3]
```diff
@@ -187,13 +185,33 @@ impl DirectoryStack {
...
+            variables.set_var("OLDPWD", "?");
+        } else {
+            variables.set_var("OLDPWD", &old_pwd);
+        }
+
+        // Update $PWD
+        let current_dir = current_dir();
+        if current_dir.is_ok() {
```
Comment (mmstick) [inline src/lib/shell/directory_stack.rs]: Why not a `match current_dir()`?

**Comment by mmstick** [comment]
Thanks!

**Pull request merged**